src/internal/m365/collection/exchange/container_resolver_test.go

```diff
@@ -1017,6 +1019,210 @@ func (suite *ConfiguredFolderCacheUnitSuite) TestAddToCache() {
+
+var _ containerGetter = mockEventContainerGetter{}
+
+type mockEventContainerGetter struct {
+	// containerGetter returns graph.CalendarDisplayable, unlike containersEnumerator
+	// which returns models.Calendarable.
+	idToCalendar map[string]graph.CalendarDisplayable
+	err          error
+}
+
+func (m mockEventContainerGetter) GetContainerByID(
+	ctx context.Context,
+	userID string,
+	dirID string,
+) (graph.Container, error) {
+	return m.idToCalendar[dirID], m.err
+}
+
+var _ containersEnumerator[models.Calendarable] = mockEventContainersEnumerator{}
+
+type mockEventContainersEnumerator struct {
+	containers []models.Calendarable
+	err        error
+}
+
+func (m mockEventContainersEnumerator) EnumerateContainers(
+	ctx context.Context,
+	userID string,
+	baseDirID string,
+) ([]models.Calendarable, error) {
+	return m.containers, m.err
+}
+
+type EventsContainerUnitSuite struct {
+	tester.Suite
+}
+
+func TestEventsContainerUnitSuite(t *testing.T) {
+	suite.Run(t, &EventsContainerUnitSuite{
+		Suite: tester.NewUnitSuite(t),
+	})
+}
+
+func makeCalendar(
+	id, name, ownerEmail string,
+	isDefault bool,
+) *models.Calendar {
+	c := models.NewCalendar()
+
+	c.SetId(ptr.To(id))
+	c.SetName(ptr.To(name))
+	c.SetIsDefaultCalendar(ptr.To(isDefault))
+
+	if len(ownerEmail) > 0 {
+		email := models.NewEmailAddress()
+
+		email.SetAddress(ptr.To(ownerEmail))
+		// Set crc as the name for keeping this func simple.
+		eName := fmt.Sprintf("%d", crc32.ChecksumIEEE([]byte(ownerEmail)))
+		email.SetName(ptr.To(eName))
+		c.SetOwner(email)
+	}
+
+	return c
+}
+
+// Test if we skip backup of shared calendars. These will be backed up for
+// the resource owner that owns the calendar.
+func (suite *EventsContainerUnitSuite) TestPopulate_SkipSharedCalendars() {
+	// map of calendars
+	calendars := map[string]models.Calendarable{
+		// Default calendars Dx
+		"D0": makeCalendar(api.DefaultCalendar, api.DefaultCalendar, "owner@bar.com", true),
+		// Atypical, but creating another default calendar for testing purposes.
+		"D1": makeCalendar("D1", "D1", "owner@bar.com", true),
+		// Shared calendars Sx
+		"S0": makeCalendar("S0", "S0", "sharer@bar.com", false),
+		// Owned calendars, not default Ox
+		"O0": makeCalendar("O0", "O0", "owner@bar.com", false),
+		// Calendars with missing owner informaton
+		"M0": makeCalendar("M0", "M0", "", false),
+	}
+
+	// Always return default calendar from the getter.
+	getContainersByID := func() map[string]graph.CalendarDisplayable {
+		return map[string]graph.CalendarDisplayable{
+			api.DefaultCalendar: *graph.CreateCalendarDisplayable(calendars["D0"], "parentID"),
+		}
+	}
+
+	table := []struct {
+		name                string
+		enumerateContainers func() []models.Calendarable
+		expectErr           assert.ErrorAssertionFunc
+		assertFunc          func(t *testing.T, ecc *eventContainerCache)
+	}{
+		{
+			name: "one default calendar, one shared",
```

Comment:
excellent set of test cases! To slightly expand this, could you please add two other tests with the following:
* default calendar has owner info set to nil and there's also a calendar the user owns and a shared calendar
* default calendar has owner info but fields in it are empty and there's also a calendar the user owns and a shared calendar

These tests will make sure we fallback correctly if there's some missing info in the default (I think it should also be ok to backup the shared calendars in this case since we're missing info). It'll also help make sure we don't have NPEs or anything as the code changes over time. Since this set of tests always sets up things to return a properly populated default calendar it may be easiest to put them in a new Test function

Reply:
Thanks @ashmrtn! I'll add these extra tests in a followup pr. issue tracking this https://github.com/alcionai/corso/issues/5208